src/test/java/tsunami/security/scanner/utilities/AppTest.java

```diff
@@ -16,12 +16,115 @@
 
 package tsunami.security.scanner.utilities;
 
+import static org.junit.Assert.assertThrows;
+import static org.mockito.Mockito.*;
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.openapi.apis.AppsV1Api;
+import io.kubernetes.client.openapi.apis.CoreV1Api;
+import io.kubernetes.client.openapi.models.V1Service;
+import io.kubernetes.client.util.Yaml;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import org.junit.Rule;
 import org.junit.Test;
-import static org.junit.Assert.*;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
 
-public class AppTest {
-    @Test public void testAppHasAGreeting() {
-        App classUnderTest = new App();
-        assertNotNull("app should have a greeting", classUnderTest.getGreeting());
-    }
+@RunWith(JUnit4.class)
+public final class AppTest {
+
+  @Rule public MockitoRule mockitoRule = MockitoJUnit.rule();
+  @Rule public TemporaryFolder folder = new TemporaryFolder();
+
+  @Mock CoreV1Api mockCoreV1Api;
+  @Mock AppsV1Api mockAppsV1Api;
+
+  String resourceConfig =
+      "apiVersion: v1\n"
+          + "kind: Service\n"
+          + "metadata:\n"
+          + "  name: jupyter\n"
+          + "  labels:\n"
+          + "    app: jupyter\n"
+          + "spec:\n"
+          + "  ports:\n"
+          + "  - port: 80\n"
+          + "    name: http\n"
+          + "    targetPort: 8888\n"
+          + "  selector:\n"
+          + "    app: jupyter\n"
+          + "  type: LoadBalancer";
+
+  @Test
+  public void runMain_whenInputValid_success() throws ApiException, IOException {
+    File jupyterFolder = folder.newFolder("jupyter");
+    File configFile = new File(jupyterFolder + "/jupyter.yaml");
+    FileWriter writer = new FileWriter(configFile);
+    writer.write(resourceConfig);
+    writer.close();
+
+    String[] args =
+        new String[] {
+          "--app",
+          "jupyter",
+          "--configPath",
+          folder.getRoot().getPath(),
+          "--templateData",
+          "{'jupyter_version':'notebook-6.0.3'}"
+        };
+
+    KubeJavaClientUtil kubeJavaClientUtil = new KubeJavaClientUtil(mockCoreV1Api, mockAppsV1Api);
+    App classUnderTest = new App(kubeJavaClientUtil);
+
+    doThrow(new RuntimeException("Skip the service creation steps."))
+        .when(mockCoreV1Api)
+        .createNamespacedService(
+            "default", (V1Service) Yaml.load(resourceConfig), null, null, null);
+
+    assertThrows(RuntimeException.class, () -> classUnderTest.run(args))
+        .getMessage()
+        .contains("Skip the service creation steps.");
```

Comment:
This test checks successful state. So instead of using a `doThrow`, simply verify the `createNamespacedService` method is called with expected argument.